resources/assets/less/functions.less

```diff
@@ -59,16 +59,11 @@
 .fragment-target-fn(@offset: 0) {
   position: static; // relative will still block pointer-events
 
-  &[id]::before {
-    content: '';
-    display: block;
-    height: @nav2-height--pinned + @offset;
-    margin: -(@nav2-height--pinned + @offset) 0 0;
-    pointer-events: none;
+  &[id] {
+    scroll-margin-top: @nav2-height--pinned + @offset;
```

Comment:
see if removing this function entirely and setting the scroll-margin-top globally instead is better

(also it's only supported in safari since last april so I'm not sure if we can migrate just yet)

Reply:
By you mean 'globally' is using wildcard `*` or what?

Hmm, only recent safari version support it :thinking:

Reply:
yes, with wildcard

Reply:
Yeah, it works correctly using wildcard.

But we still need offset support for something like this.

https://github.com/ppy/osu-web/blob/d3f74531023a46126490c4eac03f8679f1dbc207/resources/assets/less/bem/ranking-page.less#L11

I guess add variable `--scroll-margin-top-offset` and set it locally should works?